GraphColoring/src/coloring_bench.cpp

```diff
@@ -0,0 +1,57 @@
+#include <iostream>
+#include <cstdlib>
+#include <vector>
+#include <fstream>
+#include <unordered_map>
+
+#include "graph.h"
+
+#include "benchmark/benchmark.h"
+
+static const std::unordered_map<std::size_t, std::string> trace_mapping = {
+    {4, "../GraphColoring/traces/input_fc_v4.txt"},
+    {5, "../GraphColoring/traces/input_fc_v5.txt"},
+    {6, "../GraphColoring/traces/input_fc_v6.txt"},
+    {7, "../GraphColoring/traces/input_fc_v7.txt"},
+    {8, "../GraphColoring/traces/input_fc_v8.txt"},
+    {9, "../GraphColoring/traces/input_fc_v9.txt"},
+    {10, "../GraphColoring/traces/input_fc_v10.txt"}
+};
+
+static int get_chromatic_number(std::size_t num_vertices, bool use_brute_force = false)
+{
+    const std::string trace = trace_mapping.at(num_vertices);
+
+    std::ifstream ifs(trace);
+
+    Graph graph;
+    ifs >> graph;
```

Comment:
May be move some initialization of data before measurements?

Reply:
What do you mean? Graph data is read from a file

Reply:
I mean you have some data initialization (like malloc) for graph before algorithm, even if it is read from source. It can have not similar impact on resulted measurements for speed of algorithm, as it is system work to allocate space!?
As I understang google benchmark, you can move creation of graph before for(...) measurements, so it has no impact on measurements of speed

Reply:
@denislos, have a look at fixtures, they are easy to use and solve your issue just fine.

Reply:
Here's a link: https://github.com/google/benchmark#fixtures